table/tables/partition.go

```diff
@@ -313,7 +313,11 @@ func (t *partitionedTable) locateHashPartition(ctx sessionctx.Context, pi *model
 
 // GetPartition returns a Table, which is actually a partition.
 func (t *partitionedTable) GetPartition(pid int64) table.PhysicalTable {
-	return t.partitions[pid]
+	p, ok := t.partitions[pid]
```

Comment:
eg:
```go
type partition struct {
	tableCommon
}

type partitionTables struct {
	ps map[int]*partition
}

func (p *partition) GetPhysicalID() int {
	return 0
}

type tableCommon struct {
	tableID int64
}

type physibleTable interface {
	GetPhysicalID() int
}

func (pt *partitionTables) getPartition(id int) physibleTable {
	return pt.ps[id]
}

func (pt *partitionTables) getPartition2(id int) physibleTable {
	p, ok := pt.ps[id]
	if !ok {
		return nil
	}
	return p
}

func main() {
	ps := make(map[int]*partition)
	pt := &partitionTables{ps: ps}
	fmt.Println(pt.getPartition(0) == nil)  // false
	fmt.Println(pt.getPartition2(0) == nil) // true
}
```

Reply:
Golang traps and pitfalls...

A nil of type `*partition` is a kind of physibleTable

Reply:
Good catch! @crazycs520

Reply:
Would like to add some comment here?

Reply:
done.